### PR TITLE
Add docs for no-operation (empty) applies.

### PR DESCRIPTION
**Pull request by sebasslash**
Docs for empty applies!

### CODE REVIEW

File: content/cloud-docs/run/ui.mdx
Comment (Uk1288) [inline content/cloud-docs/run/ui.mdx:58]: Should this be, `...This is particularly useful...`

File: content/cloud-docs/api-docs/run.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/run.mdx:112]: don't forget :)

File: content/cloud-docs/workspaces/state.mdx
Comment (nfagerlund) [inline content/cloud-docs/workspaces/state.mdx:46]: Hmm, this is great but might benefit from a bit of expansion. 

I think the key points to try and fit into here somehow are: 

- Sometimes the state format changes between Terraform versions.
- Terraform will automatically perform any needed state upgrades when you apply a run. 
- If you choose a version of Terraform that isn't compatible with your existing state, it will fail the run and tell you which version you need to apply with first. You can solve the problem by switching to that version and running a full apply.
- You can use a no-op (empty) apply to perform these state upgrades without needing to make any config changes.

**Review by nfagerlund — CHANGES_REQUESTED** [review]
I think this is heading in a good direction, but I have some requests around the wording! 

As best as my memory serves, I _think_ you managed to find all the most important places that need to be updated for this! But I've been out of the game for a while and my Total Docs Mind-Map has crumbled a bit, so I defer to @laurapacilio to think of any other places that might want a note. 😅

**Comment by laurapacilio** [comment]
@sebasslash - I'm going to wait until you respond to @nfagerlund's comments and then I'd love to take a pass :-)

File: content/cloud-docs/api-docs/run.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/run.mdx:112]: I like that we're using "no-operation" to help people understand what this means, but the actual type is called "empty apply" so I don't want to use "no-operation" as a link. What do you think of this?

```suggestion
| `data.attributes.allow-empty-apply`                | bool                 | false                                              | Whether this run should execute a no-operation apply from a plan with no changes. Refer to [empty apply](/cloud-docs/run/modes-and-options#no-operation-apply) for details.                                                                                                                                            |
```

File: content/cloud-docs/workspaces/state.mdx
Comment (laurapacilio) [reply content/cloud-docs/workspaces/state.mdx:48]: TBD pending some discussions we're having :)

File: content/cloud-docs/run/ui.mdx
Comment (laurapacilio) [inline content/cloud-docs/run/ui.mdx:58]:
```suggestion
By default, Terraform Cloud automatically marks a run as completed if the plan contains no changes. Creating a run in [Allow Empty Apply Mode](/cloud-docs/run/modes-and-options#allow-empty-apply) allows Terraform to perform a no-operation apply from a plan containing no infrastructure changes. During an empty apply, Terraform upgrades the state file format to match the Terraform version, if required. This is particularly useful for upgrading between Terraform versions in your workspace. Refer to [upgrading state](/cloud-docs/workspaces/state#upgrading-state) for more details..  
```

**Review by laurapacilio — APPROVED** [review]
Approving so I don't block this, as I'm out tomorrow, but I have the first round of suggested edits. Some things that I think we may want to add still (pending ongoing discussions)
- Should we explain to users that they can't just upgrade more than one version at a time? My understanding is that for example, if you want to upgrade from v0.11 to v0.13, you need to upgrade to v0.12 first, run the empty apply and then upgrade to v0.13 and then run empty apply again to upgrade state.
- When do you actually get an error? I'm confused a tiny bit by this concept of "when you try to run with a versio that's not compatible with your state, you get an error" but then we say that you need to do a full apply to upgrade.

**Comment by brandonc** [comment]
@sebasslash I'm sorry to do this, but there was so much overlap between plan-only and allow-empty-apply that I decided to snapshot this PR and incorporate it into [#2313](https://github.com/hashicorp/terraform-website/pull/2313) -- including Laura's latest suggestion but removing the `workspaces/state.mdx` changes and instead going with more of an illustrated upgrade guide in `run/ui.mdx`

**Pull request closed without merging**
